### ISSUE TITLE
`LoadKeyboardLayoutA` return value bug

### ISSUE THREAD

**Issue by ibeckermayer** [microsoft/windows-rs#3178]
### Summary

One would expect this code to return an `Ok(HKL)`, however instead it behaves as shown below. Nevertheless, it does appear to load the keyboard layout as directed.

### Crate manifest

```TOML
windows-core = { version = "0.58" }
windows = { version = "0.58", features = [
    "implement",
    "Foundation_Collections",
    "UI_Input_Preview_Injection",
    "UI_UIAutomation",
    "Win32_UI_Accessibility",
    "Win32_UI_Input_KeyboardAndMouse",
    "Win32_System",
    "Win32_System_Com",
] }
```

### Crate code

```Rust
use windows::Win32::{
    Foundation::S_OK,
    UI::Input::KeyboardAndMouse::{
        LoadKeyboardLayoutA, ToUnicode, ACTIVATE_KEYBOARD_LAYOUT_FLAGS, KLF_ACTIVATE,
        KLF_REORDER, KLF_REPLACELANG, VIRTUAL_KEY, VK_CONTROL, VK_MENU, VK_SHIFT,
    },
};
use windows_core::PCSTR;

fn load_us_en_layout() {
    unsafe {
        // This api behaves weirdly, it's probably a bug in the rust translation layer. Here we take
        // an Err(e) with e.code() == S_OK to mean "success", and anything else will panic.
        match LoadKeyboardLayoutA(
            PCSTR::from_raw("00000409\0".as_ptr()), // us-en
            ACTIVATE_KEYBOARD_LAYOUT_FLAGS(KLF_ACTIVATE.0 | KLF_REORDER.0 | KLF_REPLACELANG.0),
        ) {
            Ok(_) => panic!("LoadKeyboardLayoutA should fail"),
            Err(e) => {
                assert_eq!(e.code(), S_OK);
                assert_eq!(e.message(), "The operation completed successfully.");
            }
        }
    }
}
```

**Comment by riverar** [comment]
I can't reproduce this. What OS build/version are you on?

**Comment by ibeckermayer** [comment]
@riverar thanks for trying it out. Interestingly, retrying it here on Monday and it's not reproducing for me either... but unless I was completely hallucinating, it was on Friday when I posted this.

I did find at least one individual who was experiencing a problem in a related API which resolved itself with a reboot: https://stackoverflow.com/a/10463945. For the sake of my own sanity I'm going to chalk it up to that for now. We have this call in a test suite now so I'll keep an eye out for this causing further issues.
